### PR TITLE
Danger Update: Meteors, Rods, Flux and Grav

### LINKED ISSUE #7117: Immovable Rods no longer cut through the station (again)
We had this issue before, and I fixed it. But it looks like its back again.

Instead of cutting right through the station like they should, they will often be deleted sometime before they make any significant progress. Quite disappointing.

### PR DESCRIPTION
**Pull request by TheVekter**
I think we can all agree that the station is a little less than exciting most rounds. I'm planning a series of changes to make some of the random events a little more distressing.

My design philosophy for this is as follows:
- Make most anomalies something that needs to be responded to, but won't fuck the station if it's ignored
- Make meteor waves more deadly in general (most of them just ding up the hull a bit, fuck that)
- Fix the Immovable Rod bullshit (more on that in a few)

More coming soon, but here's what we have so far:
- Meteors have been tweaked: When the station rolls for a meteor wave, it will have a 50% chance of normal, a 40% chance of dangerous, and a 10% chance of catastrophic. Also: Messed around with screen shake a bit, by request.
- Immovable Rods used to be a fair annoyance, then someone fucked with ex_act and it broke it somehow. Fixed immovable rods somehow blowing up after 20 or so tiles. Rods are still near lethal if they hit you, so be careful. Fixes #7117.
- Flux anomalies now explode much more violently - responding to them will be important unless you feel like a full department being without power.
- Gravitational anomalies now throw stuff at a valid mob in a certain radius. Should be just enough to make you get in and get out damn fast. Still bounces you if you touch it, so take care.

Unrelated:
- Fixed the Meaty Ore Wave announcement capitalization being fucked (Was 'Oh Crap, Get The Mop', is now capitalized properly).

### CODE REVIEW

**Comment by PKPenguin321** [comment]
>Bluespace: I'd like to make bluespace anoms actually move anchored shit around. Right now it just teleports items and people, and that's boring as shit.

oh boy oh boy

**Comment by TheVekter** [comment]
No shit dude. It might cause a little bit of chaos when it teleports
something interesting but otherwise there's no reason for it to take that
long to pop when it does jack shit.

File: code/modules/events/meateor_wave.dm [hunk 1]
```diff
@@ -5,7 +5,7 @@
 	max_occurrences = 1
 
 /datum/round_event/meteor_wave/meaty/announce()
-	priority_announce("Meaty ores have been detected on collision course with the station.", "Oh Crap, Get The Mop.",'sound/AI/meteors.ogg')
+	priority_announce("Meaty ores have been detected on collision course with the station.", "Oh crap, get the mop.",'sound/AI/meteors.ogg')
```
Comment (Anonus) [inline code/modules/events/meateor_wave.dm]: This is actually inconsistent with other announcements so it'll look weird and somebody will revert it a few months later.

**Comment by Razharas** [comment]
Anchored shit is excluded from bluespace one because it breaks shit with atmos and power and stuff

File: code/modules/events/meateor_wave.dm (hunk 1 above)
Comment (TheVekter) [inline code/modules/events/meateor_wave.dm]: Rabble rabble grammar rabble someone else can revert it later if they don't like it

**Comment by Anonus** [comment]
We have procs to fix that! You can call destroy on power and atmos objects and then move them instead of just letting it move everything.

**Comment by TheVekter** [comment]
Need to figure out what we want to do to make it more interesting then. If I could get it to move just tables and machinery I'd be very happy.

**Comment by palpatine213** [comment]
Wait. So with the vortex one, if you touch it while trying to scan it, do you get eaten? How the hell are you supposed to defuse it then?

**Comment by TheVekter** [comment]
No, I was under the impression it flung things like the singulo does. I'll
fuck with it tomorrow.

**Comment by Lumipharon** [comment]
Grav anomaly already throws people (it can also chain stun/throw you into a wall and kill you in like 5 seconds)
If the bluespace thing gets redone, would it be possible to bring back ye olde bluespace slime teleport reaction?

And last of all, if you're touching meteors, especially meaty ores, can you please, PLEASE make them not SHAKE THE WHOLE STATION for several minutes.
If bombs going off and a singularity eating the station don't make everything shake, then a lump of flying meat shouldn't either.

**Comment by Iamgoofball** [comment]
@Lumipharon maybe instead we should make those 2 things shake the station

**Comment by TheVekter** [comment]
I'm gonna test grav anomalies a bit more before I actually touch them.

I think we're already doing something with bluespace slimes, but I'll see
what I can do.

I'll add that to the list. I might be able to screw with the screen shake
on normal meteors too, because that shit gets annoying.

**Comment by WJohn** [comment]
Fixes #7117 then?

**Comment by TheVekter** [comment]
Should. I tested it a few times with the event and it looks like it's doing what it's supposed to do again. FINALLY.

**Comment by WJohn** [comment]
DNM on request.

**Comment by KorPhaeron** [comment]
Please remove the shaking its obnoxious and gives me motion sickness. Cool stuff in general

**Comment by AlexanderUlanH** [comment]
Changing events is all well and good, but if the goal is to make events an important part of round progression, they should be more consistent.  Maybe with separate timers and schedules for minor/normal/major events, with minor events like light outages occurring more often at the beginning of a round, some moderate events light meteors and anomalies showing up 30-40 minutes in, and consistent pressure for a shuttle call in the form of severe events after the hour marker.  The current system's supposed to provide for that by unlocking severe events later in the round, but there are so many trivial high-weight events that you still rarely see events that actually challenge or involve the crew.

**Comment by TheVekter** [comment]
I don't really intend to completely overhaul the event system, just make the station a bit more deadly.

Meteor already rolls so rarely during a round without us hitting a button that I'm not too concerned with it. Anoms happen a fair bit, so the idea is to make them not be catastrophic, but be an issue if you don't do anything about them.

**Comment by Kromgar** [comment]
Wait Immovable rods have been broken for 5 months? Wow. It felt like I hadn't seen them in a long time I missed the rod ripping a hole through the station

**Comment by Dorsisdwarf** [comment]
Meteors are an every-other-round occurrence on Basil

**Comment by TheVekter** [comment]
Right, but you have to remember that basil rounds last a good bit longer than sybil ones.

There's literally no way for me to set it up so meteors only happen every x rounds. At least, not one I'm familiar with.

**Comment by TheVekter** [comment]
This is ready to go for now. I'll work on the other stuff in a separate PR.

**Comment by WJohn** [comment]
If someone is going to give events a classification (light, moderate, dangerous) and a timer to unlock them all, they definitely should be config options.

**Comment by Miauw62** [comment]
meteors already do a lot of damage tho

**Comment by Anonus** [comment]
With the grav anomaly throwing stuff it would probably be interesting(but horribly deadly) to have it rip up floor tiles.

**Comment by TheVekter** [comment]
@Miauw62 They don't come up often enough (unless we slap the button) for it to be a major issue. All this does is make it about 5% more likely a catastrophic meteor strike will happen when the game rolls for one.

@WJohn: Not my intention (at least not yet). Meteors are the only ones coded like that.

@Anonus: Ehhh... It throwing stuff already makes it hard enough to analyze it. I want to leave Vortex as the major "oh fuck yeah no" anomaly.

File: code/modules/events/immovable_rod.dm [hunk 2]
```diff
@@ -60,7 +63,10 @@ In my current plan for it, 'solid' will be defined as anything with density == 1
 			clong.ex_act(2)
 
 	else if (istype(clong, /mob))
+		if(istype(clong, /mob/living/carbon/human))
+			var/mob/living/carbon/human/H = clong
+			H << "The rod penetrates you!"
```
Comment (RemieRichards) [inline code/modules/events/immovable_rod.dm]: should be <span class='userdanger'>

Comment (TheVekter) [inline code/modules/events/immovable_rod.dm]: What, line 68?

File: html/changelogs/vekter - 10092.txt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+
+author: Vekter
```
Comment (RemieRichards) [inline html/changelogs/vekter - 10092.txt]: this needs to be a ```.yml``` file not a ```.txt```

Comment (TheVekter) [inline html/changelogs/vekter - 10092.txt]: I'm a fuckin' idiot I'll fix it

File: code/modules/events/immovable_rod.dm (hunk 2 above)
Comment (RemieRichards) [inline code/modules/events/immovable_rod.dm]: Yes, it is a feedback message and the user is in extreme danger.
1. feedback messages relating to damage should not be in pure black (no spans)
2. feedback messages relating to the user taking damage or being in danger should use ```<span class='userdanger'>``` around the text.

File: code/modules/events/immovable_rod.dm [hunk 4]
```diff
@@ -47,6 +47,9 @@ In my current plan for it, 'solid' will be defined as anything with density == 1
 		qdel(src)
 	return ..()
 
+/obj/effect/immovablerod/ex_act(test)
+	return 0
+
 /obj/effect/immovablerod/Bump(atom/clong)
 	playsound(src, 'sound/effects/bang.ogg', 50, 1)
```
Comment (optimumtact) [inline code/modules/events/immovable_rod.dm]: can this be slowed down or limited to every second or third bump or something

It's really annoying to get 15 clangs before the rod goes out of your vision and it spams output

File: html/changelogs/vekter - 10092.yml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+
+author: Vekter
+
+delete-after: True
+
```
Comment (RemieRichards) [inline html/changelogs/vekter - 10092.yml]: need to add ```changes:``` to this line or a line above your ```-bugfix``` ```-tweak``` etc.

File: code/game/objects/effects/anomalies.dm [hunk 6]
```diff
@@ -60,6 +60,15 @@
...
 	for(var/mob/living/M in orange(4, src))
 		step_towards(M,src)
+	for(var/obj/O in orange(1,src))
+		if(!O.anchored)
+			var/mob/living/target = locate() in view(10,src)
+			if(!target)
+				return
+			else
+				O.throw_at(target, 5, 10)
+		return
```
Comment (RemieRichards) [inline code/game/objects/effects/anomalies.dm]: did you mean to ```return``` here?

this code will only ever throw one object at one mob.
and if the object is anchored it wont throw any at all.

File: code/game/objects/effects/anomalies.dm [hunk 7]
```diff
@@ -60,6 +60,15 @@
 			step_towards(O,src)
 	for(var/mob/living/M in orange(4, src))
 		step_towards(M,src)
+	for(var/obj/O in orange(1,src))
+		if(!O.anchored)
+			var/mob/living/target = locate() in view(10,src)
+			if(!target)
+				return
```
Comment (RemieRichards) [inline code/game/objects/effects/anomalies.dm]: this return is also bad,  see below.

File: html/changelogs/vekter - 10092.yml (hunk 5 above)
Comment (TheVekter) [inline html/changelogs/vekter - 10092.yml]: fucking shit ass cock piss

File: html/changelogs/vekter - 10092.yml [hunk 8]
```diff
@@ -0,0 +1,9 @@
...
+changes:
+  - bugfix: "Finally fixed immovable rods! They will now properly bisect the station (and crit anyone unlucky enough to be in their path)."
+  - tweak: "Increased Meteor activity detected in the vicinity of the station. Be advised that many storms may be more violent than they once were."
+  - rscadd: "Gravitational anomalies will now throw items at mobs in their vicinity. Take care while scanning them."
+  - rscadd: "Flux anomalies explode much more violently now. Get there and destroy them, or risk losing half a department.
```
Comment (optimumtact) [inline html/changelogs/vekter - 10092.yml]: missing end "

Comment (RemieRichards) [inline html/changelogs/vekter - 10092.yml]: rofl.

**Comment by RemieRichards** [comment]
your commit logs make me sad.

```
ermahgerd chenglerg

hurr fucking durr changelog fix

fixed the changelog FOR REAL THIS TIME, also fixed bad coding

FUCK
```

It's fine though.

File: code/modules/events/immovable_rod.dm (hunk 2 above)
Comment (Iamgoofball) [inline code/modules/events/immovable_rod.dm]: uhm

i didn't expect that to work

i theorycrafted this as a joke in piratepad

**Pull request merged**